store/src/java-test/com/zimbra/cs/util/JWTBasedAuthTest.java

```diff
@@ -149,4 +150,17 @@ public void testGetSaltCookie() {
         String saltFound = JWTUtil.getSalt(null, engineCtxt);
         Assert.assertEquals(salt, saltFound);
     }
+
+    @Test
+    public void testClearSalt() {
+        String salt = "s1";
+        String zmJWTCookieValue = "s2|s3|s1";
+        Assert.assertEquals("s2|s3", JWTUtil.clearSalt(zmJWTCookieValue, salt));
+        zmJWTCookieValue = "s2|s1|s3";
+        Assert.assertEquals("s2|s3", JWTUtil.clearSalt(zmJWTCookieValue, salt));
+        zmJWTCookieValue = "s1|s2|s3";
+        Assert.assertEquals("s2|s3", JWTUtil.clearSalt(zmJWTCookieValue, salt));
+        zmJWTCookieValue = "s1";
+        Assert.assertEquals("", JWTUtil.clearSalt(zmJWTCookieValue, salt));
+    }
```

Comment:
For completeness also add the case where you try to remove a non existing salt.

Reply:
done.